File: iteration2/CourseRegistrationSystem/src/Course.java
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ArrayNode;
import com.fasterxml.jackson.databind.node.ObjectNode;

import java.io.File;
import java.io.IOException;

public class Course{

    private String courseId;
    private String courseName;
    private int credit;
    @JsonProperty("prerequisite")
    private boolean prerequisite;
    private String prerequisiteLessonId;
    private CourseSection courseSection;
    
    

    public Course(String CourseId, String CourseName, int credit, boolean prerequisite, String prerequisiteLessonId, CourseSection courseSection){
        this.courseId = CourseId;
        this.courseName = CourseName;
        this.credit = credit;
        this.prerequisite = prerequisite;
        this.prerequisiteLessonId = prerequisiteLessonId;
        this.courseSection = courseSection;
    }

    public Course(){}

    

    //public List<Student> getEnrolledStudents(){
    //  
    //}

    public boolean addEnrolledStudent(Student student, Course course){
        JSONMethods jM = new JSONMethods();
        return jM.addEnrolledStudent(student,course);
    }

    public boolean hasPrerequisite(){
        return prerequisite;
    }

    public String toString() {
        StringBuilder sb = new StringBuilder();

         // Course Information
        String courseInfo = String.format("%-18s %-40s %-8s %-20s %-10s %-23s %-20s","   "+
                courseId, courseName, credit, prerequisiteLessonId, courseSection.getTerm(),courseSection.getDay()+"-"+ courseSection.getHour(),courseSection.getInstructor());
        sb.append(courseInfo);

        return sb.toString();
    }


    public String getCourseId(){
        return courseId;
    }

    public int getCredit(){
        return credit;
    }

    public String getCourseName(){
        return courseName;
    }

    public String getPrerequisiteLessonId(){
        return prerequisiteLessonId;
    }


    public CourseSection getCourseSection(){
        return courseSection;
    }

}

File: iteration3/Course.py
from JsonMethods import JSONMethods


class Course:
    def __init__(self, course_id="", course_name="", credit=0, prerequisite=False, prerequisite_lesson_id="", course_section=None):
        self.course_id = course_id
        self.course_name = course_name
        self.credit = credit
        self.prerequisite = prerequisite
        self.prerequisite_lesson_id = prerequisite_lesson_id
        self.course_section = course_section

    def add_enrolled_student(self, student, course):
        # Assuming JSONMethods is handling the enrollment process
        jM = JSONMethods()
        return jM.add_enrolled_student(student, course)

    def has_prerequisite(self):
        return self.prerequisite

    def __str__(self):
        course_info = f"{self.course_id:<18} {self.course_name:<40} {self.credit:<8} {self.prerequisite_lesson_id:<20} {self.course_section.term:<10} {self.course_section.day}-{self.course_section.hour:<23} {self.course_section.instructor:<20}"
        return course_info

    def get_course_id(self):
        return self.course_id

    def get_credit(self):
        return self.credit

    def get_course_name(self):
        return self.course_name

    def get_prerequisite_lesson_id(self):
        return self.prerequisite_lesson_id

    def get_course_section(self):
        return self.course_section
